### PR TITLE
add lazy-load-module rule

### PR DESCRIPTION
**Pull request by inkz**
Simple rule for bad pattern: dynamic `require` usage

### CODE REVIEW

File: javascript/lang/best-practice/lazy-load-module.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+rules:
+- id: lazy-load-module
+  patterns:
+  - pattern: require(...)
+  - pattern-inside: |
+      function $NAME(...) {
+          ...
+      }
+  message: |
+    Lazy loading can complicate code bundling if care is not taken.
```
Comment (minusworld) [inline javascript/lang/best-practice/lazy-load-module.yaml]: What's the recommended fix for this? Just to do global requiring?

Comment (inkz) [reply javascript/lang/best-practice/lazy-load-module.yaml]: Yes! and I definitely need to add this information to the message

Comment (inkz) [reply javascript/lang/best-practice/lazy-load-module.yaml]: done!

**Review by minusworld — APPROVED** [review]
excellent recommendation :D

**Pull request merged**